client/src/components/Modal/Modal.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import "./Modal.css";
+
+
+// Class Based React Component
+class Modal extends Component {
+  constructor(props) {
+    super(props);
+    console.log(props);
+
+    // Default CSS class to apply to the Component
+    this.state = {
+      classList: "Modal",
+    };
+  }
+  // Runs after Component is loaded in the broswer
+  componentDidMount() { }
+  render() {
+    if (this.props.join) {
+      return (
+        <div className={this.state.classList}>
+          <div className="modal-header">
+            <span className="close-modal-btn">×</span>
+          </div>
+          <div className="modal-body">
+            <div className="htag">
+              <h3>Are you sure you want to join {this.props.children}?</h3>
+            </div>
+          </div>
+          <div className="modal-footer">
+            <div className="cancel">
+              <button className="cancel">Cancel</button>
+            </div>
+            <div className="confirm">
+              <button className="confirm">Join</button>
+            </div>
+          </div>
+        </div>
+      )
+    }
+    return (
+      <div className={this.state.classList}>
+        <div className="modal-header">
+          <span className="close-modal-btn">×</span>
+        </div>
+        <div className="modal-body">
+          <div className="htag">
+            <h3>Are you sure you want to follow {this.props.children}?</h3>
+          </div>
+        </div>
+        <div className="modal-footer">
+          <div className="cancel">
+            <button className="cancel">Cancel</button>
+          </div>
+          <div className="confirm">
+            <button className="confirm">Confirm</button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
```

Comment:
You can shorten this up a lot - this is a lot of duplication. The text of the button can come in as a prop and then. Remember also that you want to accept props for what happens when someone clicks on cancel, the X, or confirm!